src/menus/icon-menu.jsx

```diff
@@ -167,32 +177,33 @@ const IconMenu = React.createClass({
   },
 
   close(isKeyboard) {
-    if (this.state.open) {
-      this.setState({open: false}, () => {
-        //Set focus on the icon button when the menu close
-        if (isKeyboard) {
-          let iconButton = this.refs[this.state.iconButtonRef];
-          ReactDOM.findDOMNode(iconButton).focus();
-          iconButton.setKeyboardFocus();
-        }
-      });
+    if (!this.state.open) {
+      return;
     }
+    this.setState({open: false}, () => {
+      //Set focus on the icon button when the menu close
+      if (isKeyboard) {
+        let iconButton = this.refs[this.state.iconButtonRef];
+        ReactDOM.findDOMNode(iconButton).focus();
+        iconButton.setKeyboardFocus();
+      }
+    });
   },
 
-  open(menuInitiallyKeyboardFocused) {
-    if (!this.state.open) {
-      this.setState({
-        open: true,
-        menuInitiallyKeyboardFocused: menuInitiallyKeyboardFocused,
-      });
-    }
+  open(menuInitiallyKeyboardFocused, e) {
+    this.setState({
+      open: true,
+      menuInitiallyKeyboardFocused: menuInitiallyKeyboardFocused,
+      anchorEl: e.currentTarget,
+    });
+    e.preventDefault();
   },
 
   _handleItemTouchTap(e, child) {
-
     if (this.props.closeOnItemTouchTap) {
       let isKeyboard = Events.isKeyboard(e);
 
+
       this._timeout = setTimeout(() => {
         this.close(isKeyboard);
```

Comment:
With the timeout, I think that we should check that the component is still mounted.